plugins/flytekit-pandera/flytekitplugins/pandera/pandas_transformer.py

```diff
@@ -0,0 +1,144 @@
+    def __init__(self):
+        super().__init__("Pandera Transformer", pandera.typing.DataFrame)  # type: ignore
+        self._sd_transformer = StructuredDatasetTransformerEngine()
+
+    def _get_pandera_schema(self, t: Type[pandera.typing.DataFrame]):
+        config = ValidationConfig()
+        if typing.get_origin(t) is typing.Annotated:
+            t, *args = typing.get_args(t)
+            # get pandera config
+            for arg in args:
+                if isinstance(arg, ValidationConfig):
+                    config = arg
+                    break
+
+        try:
+            type_args = typing.get_args(t)
+        except AttributeError:
+            # for python < 3.8
+            type_args = getattr(t, "__args__", None)
+
+        if type_args:
+            schema_model, *_ = type_args
+            schema = schema_model.to_schema()
+        else:
+            schema = pandera.DataFrameSchema()  # type: ignore
+        return schema, config
+
+    @staticmethod
+    def _get_pandas_type(pandera_dtype: pandera.dtypes.DataType):
+        return pandera_dtype.type.type
+
+    def _get_col_dtypes(self, t: Type[pandera.typing.DataFrame]):
+        schema, _ = self._get_pandera_schema(t)
+        return {k: self._get_pandas_type(v.dtype) for k, v in schema.columns.items()}
+
+    def get_literal_type(self, t: Type[pandera.typing.DataFrame]) -> LiteralType:
+        if typing.get_origin(t) is typing.Annotated:
+            t, _ = typing.get_args(t)
+        return self._sd_transformer.get_literal_type(t)
+
+    def assert_type(self, t: Type[T], v: T):
+        if not hasattr(t, "__origin__") and not isinstance(v, (t, pandas.DataFrame)):
+            raise TypeError(f"Type of Val '{v}' is not an instance of {t}")
+
+    def to_literal(
+        self,
+        ctx: FlyteContext,
+        python_val: Union[pandas.DataFrame, StructuredDataset],
+        python_type: Type[pandera.typing.DataFrame],
+        expected: LiteralType,
+    ) -> Literal:
+        assert isinstance(
+            python_val, (pandas.DataFrame, StructuredDataset)
+        ), f"Only Pandas Dataframe object can be returned from a task, returned object type {type(python_val)}"
+
+        if isinstance(python_val, StructuredDataset):
+            lv = self._sd_transformer.to_literal(ctx, python_val, pandas.DataFrame, expected)
+            python_val = self._sd_transformer.to_python_value(ctx, lv, pandas.DataFrame)
+
+        schema, config = self._get_pandera_schema(python_type)
+        renderer = PandasReportRenderer(title=f"Pandera Report: {schema.name}")
+        try:
+            val = schema.validate(python_val, lazy=True)
+        except (pandera.errors.SchemaError, pandera.errors.SchemaErrors) as exc:
+            if config.on_error == "raise":
+                raise exc
+            elif config.on_error == "warn":
+                warnings.warn(str(exc), RuntimeWarning)
+                html = renderer.to_html(exc)
+                Deck(renderer._title, html)
```

Comment:
> Ideally on_error="raise" will also render a deck, but right now there's a limitation where decks aren't rendered with failed tasks.

we're fixing this in Streaming decks, so how about we future-proof this by adding the Deck in both cases?